allennlp/modules/token_embedders/embedding.py

```diff
@@ -229,9 +240,19 @@ def _read_pretrained_word2vec_format_embedding_file(embeddings_filename: str, #
 
     # First we read the embeddings from the file, only keeping vectors for the words we need.
     logger.info("Reading embeddings from file")
-    with gzip.open(cached_path(embeddings_filename), 'rb') as embeddings_file:
+
+    def read_embedding_file(url_or_path):
```

Comment:
same comment, plus I'm not sure it's worth making this a function if it's only going to get called once immediately

Reply:
I defined it mainly for readability. The call to ``open_maybe_compressed_file`` is pretty long, so I'd either do this: 
```
with open_maybe_compressed_file(local_path, mode='t',
                                encoding=_WORD2VEC_ENCODING,
                                file_format=file_format) as embedding_file:
    for line in embedding_file:
```
or this:
```
embedding_file =  open_maybe_compressed_file(local_path, mode='t',
                                             encoding=_WORD2VEC_ENCODING,
                                             file_format=file_format)
with embedding_file:
    for line in embedding_file:
```

Furthermore, the function argument allowed me to clarify that ``embedding_filename`` is not a filename but a URL or a path to the file. About it, I think ``embedding_filename`` should be renamed in the entire module.

Anyway, I can change if you prefer.

Reply:
yeah, the former was what I was imagining.